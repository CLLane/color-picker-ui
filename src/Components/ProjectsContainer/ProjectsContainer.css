.project-container__section {
  display: flex;
  justify-content: space-between;
  overflow: scroll;
}