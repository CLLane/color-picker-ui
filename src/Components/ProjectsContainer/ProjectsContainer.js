import React from 'react';
import ProjectCard from '../ProjectCard/ProjectCard'
import './ProjectsContainer.css';

export const ProjectsContainer = ({ projects, palettes }) => {
  const projectCards = projects.map((project, index) => {
    const projectPalettes = palettes.filter( palette => project.id === palette.project_id)
    return <ProjectCard project={project} palettes={projectPalettes} key={index} />
  })

  return (
    <section>
      {projectCards}
    </section>
  )
}

export default ProjectsContainer;