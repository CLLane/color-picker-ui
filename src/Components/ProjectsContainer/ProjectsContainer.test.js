import React from 'react';
import { shallow } from 'enzyme';
import './ProjectsContainer.css';
import ProjectsContainer from './ProjectsContainer';

describe('Projects Container', () => {
  let wrapper;
  let mockProjects;
  let mockPalettes;
  let mockGrabPalette;
  let mockUpdatePaletteName;
  let mockUpdateProjectName;
  let mockTrashPalette;
  let mockTrashProject;
  beforeEach(() => {
    mockPalettes = [
      {
        id: 1, 
        name: 'Cool Colors',
        project_id: 12345,
        colorOne: '#FFFFFF',
        colorTwo: '#FFFFFF',
        colorThree: '#FFFFFF',
        colorFour: '#FFFFFF',
        colorFive: '#FFFFFF'
      },
      {
        id: 2, 
        name: 'Warm Colors',
        project_id: 12345,
        colorOne: '#FFFFFF',
        colorTwo: '#FFFFFF',
        colorThree: '#FFFFFF',
        colorFour: '#FFFFFF',
        colorFive: '#FFFFFF'
      }
    ];
    mockProjects = [{
      id: 12345,
      name: 'Cool Project',
      user_id: 1
    }];
    mockTrashProject = jest.fn();
    mockTrashPalette = jest.fn();
    mockGrabPalette = jest.fn();
    mockUpdatePaletteName = jest.fn();
    wrapper = shallow(<ProjectsContainer 
      projects={mockProjects}
      palettes={mockPalettes}
      showPalette={mockGrabPalette}
      updatePaletteName={mockUpdatePaletteName}
      updateProjectName={mockUpdateProjectName}
      trashProject={mockTrashProject}
      trashPalette={mockTrashPalette}
    />);
  });

  it('should match the snapshot, given a set of colors', () => {
    expect(wrapper).toMatchSnapshot();
  });

});