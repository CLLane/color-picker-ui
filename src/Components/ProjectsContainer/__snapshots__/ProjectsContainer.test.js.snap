// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Container should match the snapshot, given a set of colors 1`] = `
<section
  className="project-container__section"
>
  <ProjectCard
    key="0"
    palettes={
      Array [
        Object {
          "colorFive": "#FFFFFF",
          "colorFour": "#FFFFFF",
          "colorOne": "#FFFFFF",
          "colorThree": "#FFFFFF",
          "colorTwo": "#FFFFFF",
          "id": 1,
          "name": "Cool Colors",
          "project_id": 12345,
        },
        Object {
          "colorFive": "#FFFFFF",
          "colorFour": "#FFFFFF",
          "colorOne": "#FFFFFF",
          "colorThree": "#FFFFFF",
          "colorTwo": "#FFFFFF",
          "id": 2,
          "name": "Warm Colors",
          "project_id": 12345,
        },
      ]
    }
    project={
      Object {
        "id": 12345,
        "name": "Cool Project",
        "user_id": 1,
      }
    }
    showPalette={[MockFunction]}
    trashPalette={[MockFunction]}
    trashProject={[MockFunction]}
    updatePaletteName={[MockFunction]}
  />
</section>
`;
